src/pages/SignUp.jsx

```diff
@@ -1,11 +1,188 @@
 import React from "react";
 
-function SignUp() {
+import { useFormik } from "formik";
+
+import Email from "../assets/Email.svg";
+import Key from "../assets/Key.svg";
+import User from "../assets/User.svg";
+import Phone from "../assets/Phone.svg";
+
+const validate = (values) => {
+  const errors = {};
+
+  if (!values.userName) {
+    errors.userName = "Required";
+  }
+  if (!values.email) {
+    errors.email = "Required";
+  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
+    errors.email = "Invalid email address";
+  }
+  if (!values.password) {
+    errors.password = "Required";
+  } else if (values.password.length < 8) {
+    errors.password = "Password should be at least 8 characters ";
+  }
+  if (!values.confirmPassword) {
+    errors.confirmPassword = "Required";
+  } else if (values.confirmPassword !== values.password) {
+    errors.confirmPassword = "Password don't match";
+  }
+
+  return errors;
+};
```

Comment:
Let's a [schema validation](https://formik.org/docs/guides/validation#validationschema) here, much more cleaner and easier to handle.